bindings/go/src/fdb/tuple/tuple.go

```diff
@@ -164,14 +164,29 @@ func (p *packer) encodeInt(i int64) {
 		binary.BigEndian.PutUint64(scratch[:], uint64(i))
 	case i < 0:
 		n = bisectLeft(uint64(-i))
-		p.putByte(byte(0x14 - n))
+		p.putByte(byte(intZeroCode - n))
 		offsetEncoded := int64(sizeLimits[n]) + i
 		binary.BigEndian.PutUint64(scratch[:], uint64(offsetEncoded))
 	}
 
 	p.putBytes(scratch[8-n:])
 }
 
+func (p *packer) encodeUint(i uint64) {
```

Comment:
The resulting LoC would be similar using a type switch though you might prefer it:
```go
func (p *packer) encodeInt(i interface{}) {
	var n int
	var scratch [8]byte

	switch i := i.(type) {
	case uint64:
		if i == 0 {
			p.putByte(intZeroCode)
			return
		}
		n = bisectLeft(i)
		p.putByte(byte(intZeroCode + n))
		binary.BigEndian.PutUint64(scratch[:], i)
	case int64:
		if i == 0 {
			p.putByte(intZeroCode)
			return
		}
		switch {
		case i > 0:
			n = bisectLeft(uint64(i))
			p.putByte(byte(intZeroCode + n))
			binary.BigEndian.PutUint64(scratch[:], uint64(i))
		case i < 0:
			n = bisectLeft(uint64(-i))
			p.putByte(byte(intZeroCode - n))
			offsetEncoded := int64(sizeLimits[n]) + i
			binary.BigEndian.PutUint64(scratch[:], uint64(offsetEncoded))
		}
	}

	p.putBytes(scratch[8-n:])
}
```

I chose the other way initially because there is already a type switch in `encodeTuple`. Just let me know and I will change them to this if you would like.

Reply:
Okay, sure. It sounds like the answer is "go's type system isn't quite strong enough to do what I want", so I guess it's fine as is.

Reply:
One option to avoid duplication would be to call `encodeUint` from `encodeInt` for all values that are greater than or equal to 0. Then each case (<0, ==0, and >0) is only written once.